src/condvar.rs

```diff
@@ -555,12 +558,20 @@ mod tests {
         let mut g = m.lock();
         let no_timeout = c.wait_for(&mut g, Duration::from_millis(1));
         assert!(no_timeout.timed_out());
+
         let _t = thread::spawn(move || {
             let _g = m2.lock();
             c2.notify_one();
         });
-        let timeout_res = c.wait_for(&mut g, Duration::from_millis(u32::max_value() as u64));
+        // Non-nightly panics on too large timeouts. Nightly treats it as indefinite wait.
+        #[cfg(feature = "nightly")]
+        let very_long_timeout = Duration::from_secs(u64::max_value());
+        #[cfg(not(feature = "nightly"))]
+        let very_long_timeout = Duration::from_millis(u32::max_value() as u64);
```

Comment:
This would look better with `if cfg!(feature = "nightly") {} else {}`.

Reply:
I instinctively always try to use `#[cfg]` to make the unwanted code go away as early as possible in the compile :)
I also think that won't work. Since `checked_add` does not exist on non-nightly. So it does not compile. `if false { checked_add() }` still requires `checked_add` even if it's never going to be executed.

Reply:
Sorry.. I misread which file the comment was on. Yes, here we can use `cfg!`. Fixed it.